packages/relay/src/lib/eth.ts

```diff
@@ -1443,12 +1444,39 @@ export class EthImpl implements Eth {
     this.logger.trace(`${requestIdPrefix} getTransactionReceipt(${hash})`);
 
     const cacheKey = `${constants.CACHE_KEY.ETH_GET_TRANSACTION_RECEIPT}_${hash}`;
-    let cachedResponse = this.cache.get(cacheKey, EthImpl.ethGetTransactionReceipt);
+    let cachedResponse = this.cache.get(cacheKey, EthImpl.ethGetTransactionReceipt, requestIdPrefix);
     if (cachedResponse) {
       this.logger.debug(`${requestIdPrefix} getTransactionReceipt returned cached response: ${cachedResponse}`);
       return cachedResponse;
     }
 
+    const cacheKeySyntheticLog = `${EthImpl.cacheKeySyntheticLogHash}${hash}`;
+    const cachedSyntheticResponse = this.cache.get(cacheKeySyntheticLog, EthImpl.ethGetTransactionReceipt, requestIdPrefix);
+    if (cachedSyntheticResponse) {
+      const cachedLog: Log = JSON.parse(cachedSyntheticResponse);
+      
+      const receipt: any  = {
+        blockHash: cachedLog.blockHash,
+        blockNumber: cachedLog.blockNumber,
+        from: null,
+        to: cachedLog.address,
+        cumulativeGasUsed: null,
+        gasUsed: null,
+        contractAddress: cachedLog.address,
+        logs: [cachedLog],
+        logsBloom: EthImpl.emptyBloom,
+        transactionHash: cachedLog.transactionHash,
+        transactionIndex: cachedLog.transactionIndex,
+        effectiveGasPrice: null,
+        root: null,
+        status: null,
+      };
+      this.logger.debug(`${requestIdPrefix} getTransactionReceipt returned cached synthetic receipt response: ${cachedResponse}`);
+      this.cache.set(cacheKey, receipt, EthImpl.ethGetTransactionReceipt, undefined, requestIdPrefix);
```

Comment:
```suggestion
      this.cache.set(cacheKey, receipt, cacheKeySyntheticLog, syntheticLogCacheTtl, requestIdPrefix);
```

Would need to add the following to the top and set default to be like 1hr

```
private readonly syntheticLogCacheTtl =  parseNumericEnvVar('SYNTHETIC_LOG_CACHE_TTL', 'DEFAULT_SYNTHETIC_LOG_CACHE_TTL'));
```